cli/src/commands/workspace/list.rs

```diff
@@ -12,27 +12,51 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use clap_complete::ArgValueCandidates;
+use jj_lib::commit::Commit;
 use jj_lib::repo::Repo as _;
 use tracing::instrument;
 
 use crate::cli_util::CommandHelper;
 use crate::command_error::CommandError;
+use crate::complete;
+use crate::templater::TemplateRenderer;
 use crate::ui::Ui;
 
 /// List workspaces
 #[derive(clap::Args, Clone, Debug)]
-pub struct WorkspaceListArgs {}
+pub struct WorkspaceListArgs {
+    /// Render each workspace using the given template
+    ///
+    /// You can specify arbitrary template expressions using the
+    /// [built-in keywords]. See [`jj help -k templates`] for more information.
+    ///
+    /// [built-in keywords]:
+    ///     https://jj-vcs.github.io/jj/latest/templates/#commit-keywords
+    ///
+    /// [`jj help -k templates`]:
+    ///     https://jj-vcs.github.io/jj/latest/templates/
+    #[arg(long, short = 'T', add = ArgValueCandidates::new(complete::template_aliases))]
+    template: Option<String>,
+}
 
 #[instrument(skip_all)]
 pub fn cmd_workspace_list(
     ui: &mut Ui,
     command: &CommandHelper,
-    _args: &WorkspaceListArgs,
+    args: &WorkspaceListArgs,
 ) -> Result<(), CommandError> {
     let workspace_command = command.workspace_helper(ui)?;
     let repo = workspace_command.repo();
+    let template: TemplateRenderer<Commit> = {
+        let language = workspace_command.commit_template_language();
+
+        match &args.template {
+            Some(value) => workspace_command.parse_template(ui, &language, value)?,
+            None => workspace_command.commit_summary_template(),
+        };
+    };
     let mut formatter = ui.stdout_formatter();
-    let template = workspace_command.commit_summary_template();
     for (name, wc_commit_id) in repo.view().wc_commit_ids() {
         write!(formatter, "{}: ", name.as_symbol())?;
```

Comment:
I think both `name` and `commit` should be templated. Naming aside, maybe we can add `WorkspaceRef` template type similar to `CommitRef`, and `commit.working_copies()` will return `List<WorkspaceRef>`.